ocw/lib/EC2.py

```diff
@@ -85,11 +86,14 @@ def all_regions(self):
         return regions
 
     def delete_instance(self, instance_id):
-        instances_list = list(self.ec2_resource().instances.filter(InstanceIds=[instance_id]))
-        if len(instances_list) > 0:
-            instances_list[0].terminate()
-        else:
-            logger.warning("Instance {} is ACTIVE in local DB but does not exists on EC2".format(instance_id))
+        try:
+            self.ec2_resource().instances.filter(InstanceIds=[instance_id]).terminate()
+        except ClientError as ex:
+            if ex.response['Error']['Code'] == 'InvalidInstanceID.NotFound':
+                logger.warning("Instance {} is ACTIVE in local DB but does not exists on EC2".format(instance_id))
```

Comment:
I am ok with this warning message, but since this function doesn't care about the DB, the reason could be different, for example you have given a wrong `instance_id` value or similar. I would say `Cannot delete instance ID {}. It does not exist on EC2` or something similar

Reply:
done